File: Chapter06/gpio-poll.php
#!/usr/bin/php
<?php
        define("gpio24", "/sys/class/gpio/gpio24/value");
        define("gpio91", "/sys/class/gpio/gpio91/value");

	# Get the GPIOs streams
	$stream24 = fopen(gpio24, 'r');
	$stream91 = fopen(gpio91, 'r');

	while (true) {
		# Set up stream sets for the select()
		$read = NULL;
		$write = NULL;
		$exept = array($stream24, $stream91);

		# Wait for IRQs (without timeout)...
		$ret = stream_select($read, $write, $exept, NULL);
		if ($ret < 0)
			die("stream_select: error");

		foreach ($exept as $input => $stream) {
			# Read the GPIO status
			fseek($stream, 0, SEEK_SET);
        		$status = intval(fgets($stream));

			# Get the filename from "/sys/class/gpio/gpioXX/value"
			$meta_data = stream_get_meta_data($stream);
			$gpio = basename(dirname($meta_data["uri"]));

			printf("$gpio status=$status\n");
		}
	}
?>

File: Chapter06/gpio-poll.py
#!/usr/bin/python

from __future__ import print_function
import os
import sys
import select

gpio24  = "/sys/class/gpio/gpio24/value"
gpio91  = "/sys/class/gpio/gpio91/value"

# Get the GPIOs streams
stream24 = open(gpio24, 'r');
stream91 = open(gpio91, 'r');

while True :
	# Set up stream sets for the select()
	read = []
	write = []
	exept = [stream24, stream91]

	# Wait for IRQs (without timeout)...
	r, w, e = select.select(read, write, exept)
	for i, input in enumerate(e) :
		# Read the GPIO status
		input.seek(0, 0)
    		status = input.read().rstrip("\n")

		# Get the filename from "/sys/class/gpio/gpioXX/value"
		path = os.path.dirname(input.name)
		gpio = os.path.basename(path)

		print("%s status=%s" % (gpio, status))
